### ISSUE TITLE
rehearse-plugin: add an optional PR number to the rehearse command

### ISSUE THREAD

**Issue by dhiller** [kubevirt/project-infra#2898]
# TL;DR

goal: add optional pull-request-number parameter to `/rehearse test-job` command

# Task

The `/rehearse` command is used to rehearse prow jobs whose definitions reside in repository `project-infra` inside the `github/ci/prow-deploy/files/jobs` folder.
This is done by the [rehearse plugin](https://github.com/kubevirt/project-infra/tree/main/external-plugins/rehearse) that is implemented as an external plugin for  [prow](https://github.com/kubernetes/test-infra/tree/master/prow#readme).

Current situation: whenever we run a rehearsal, per job created we check out either the default branch or one of the branches configured, whichever is defined in `branches:` section in the presubmit definition: https://github.com/kubevirt/project-infra/blob/main/external-plugins/rehearse/plugin/handler/handler.go#L415

As stated [here](https://github.com/kubevirt/project-infra/pull/2889#issuecomment-1643484807) a feature to select a PR instead of the default branch when running rehearsals against the jobs would definitely be very useful.

## Example

A comment on the above PR like this
```
/rehearse pull-project-infra-test-robots 2889
```
could check out the branch for that PR and test changes from the PR branch together with the new job configuration. That could also be useful for jobs of other repositories, so that you can test changes altogether.

**Comment by dhiller** [comment]
/assign

need to prepare this as a

/good-first-issue
